src/seedu/addressbook/data/person/Address.java

```diff
@@ -23,13 +35,32 @@
     public Address(String address, boolean isPrivate) throws IllegalValueException {
         String trimmedAddress = address.trim();
         this.isPrivate = isPrivate;
-        if (!isValidAddress(trimmedAddress)) {
+        if (isValidAddress(trimmedAddress)) {
             throw new IllegalValueException(MESSAGE_ADDRESS_CONSTRAINTS);
         }
-        this.value = trimmedAddress;
+        String[] splited = trimmedAddress.split(SEPARATE);
```

Comment:
Good attempt on using SLAP.

Should put the splitting in the next level method for better SLAP though.